torch/_prims_common/wrappers.py

```diff
@@ -163,8 +163,8 @@ def _safe_copy_out(
     if exact_dtype:
         utils.check(
             copy_from.dtype == copy_to.dtype,
-            lambda: f"Expected out tensor to have dtype {copy_from.dtype} "
-            f"but got {copy_to.dtype} instead",
+            lambda: (f"Expected the dtype for input and out to match, but got {copy_from.dtype} for input's dtype "
```

Comment:
I don't think it's always true that the `copy_from` datatype is the datatype of an "input." I think there are actually some cases where the "safe copy" is from an intermediate datatype to the output.

cc @lezcano

Reply:
Note that this is under `exact_dtype`. This is just used in functions that *need* the outputs to match. Think linalg functions that call external libraries.

Reply:
I think Mike's right.

I ended up using `exact_dtype` in the `complex` PR https://github.com/pytorch/pytorch/pull/88562 due to non-standard promotion semantics. There, I need to assert that `result` and `out` match exactly, but `result`'s dtype is **not** the dtype of input (float), it's an intermediate dtype (complex).

In general, strongly asserting that it's input's dtype can be misleading. So I'll revert this change.